src/filefinder.cpp

```diff
@@ -788,24 +786,28 @@ FileFinder::Directory FileFinder::GetDirectoryMembers(const std::string& path, F
 		std::string const name = ent->d_name;
 	#endif
 #endif
-		bool is_directory;
+
+		static bool has_fast_dir_stat = true;
+		bool is_directory = false;
 		if (has_fast_dir_stat) {
 			#ifdef PSP2
 			is_directory = S_ISDIR(ent.d_stat.st_mode);
+			#elif defined(_DIRENT_HAVE_D_TYPE)
+			if (ent->d_type == DT_UNKNOWN) {
+				has_fast_dir_stat = false;
```

Comment:
I could move the deleted output::debug lines here but I question there usefulness

Reply:
At least they are an indicator when porting to a new platform... but could be left out, right.